File: 0026-remove-duplicates-from-sorted-array/solution.c
int removeDuplicates(int* nums, int numsSize) {
    if (numsSize == 0) {
        return 0;
    }
    int slowpointer = 0;
    for (int fastpointer = 1; fastpointer < numsSize; fastpointer++) {
        if (nums[fastpointer] != nums[slowpointer]) {
            slowpointer++;
            nums[slowpointer] = nums[fastpointer];
        }
    }
    return slowpointer + 1;
}

File: 0026-remove-duplicates-from-sorted-array/solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums.length == 1){
            return 1;
        }

        int left_p = 0;
        int left = 0;
        int right = 0;

        while(right < nums.length && left < nums.length){
            if(nums[left] == nums[right]){
                right++;
                continue;
            }else{
                nums[left_p ++] = nums[left];
                left = right;
                right++;
            }
        }
        nums[left_p ++] = nums[left];
        return left_p;
    }
}
